### ISSUE TITLE
Integration with Skoruba Admin to replace Identity Server

### ISSUE THREAD

**Issue by weedkiller** [openiddict/openiddict-core#1111]
Hello, I did not know this existed, but great work! 💯 

They are making Identity Server a fully commercial license now, so this is a great option as suggested by @larsfjerm [here](https://github.com/skoruba/IdentityServer4.Admin/issues/718) 

Can you please help share some guidance and steps to configure openiddict-core with [GUI Admin](https://github.com/skoruba/IdentityServer4.Admin)

- Code snippets
- Documentations on where to set it up

Thanks

**Comment by kevinchalet** [comment]
Hi,

> Hello, I did not know this existed, but great work! 💯

Thanks 😄 

> Can you please help share some guidance and steps to configure openiddict-core with GUI Admin

Unfortunately, I don't think it would be that easy, as `IdentityServer4.Admin` is tied to IdentityServer4's APIs (which is not unreasonable for an IdSrv plugin 😅). Nothing would prevent you from reworking it completely to use OpenIddict's core APIs instead, but I suspect that would be a time-consuming process.

While OpenIddict doesn't have any GUI/admin panel, OrchardCore has an OpenID module based on OpenIddict, that includes a management UI for client apps. It's still quite limited, but contributions are welcome to improve that, so don't hesitate.

Cheers.

**Comment by weedkiller** [comment]
first, appreciate your response

I'm fine with reworking the code and API's to retrofit, I think many will follow as it becomes an natural/easier path.

Would we be able to reuse the Orchard Core GUI views in and ASP app easily. Happy to contribute... just need to get caught up more

**Comment by kevinchalet** [comment]
This blog post I just wrote should help make things clearer: https://kevinchalet.com/2020/10/03/using-the-orchardcore-openid-management-feature-with-an-existing-openiddict-deployment/ 😃

**Comment by papyr** [comment]
Hi I was reading your article, whats is the velucia server and any chance you can give us an API for the membership would like to try it out as well

**Comment by kevinchalet** [comment]
> Hi I was reading your article, whats is the velucia server

![image](https://user-images.githubusercontent.com/6998306/95018545-b9470b00-0660-11eb-8bda-a9a080ba4d9b.png)

> any chance you can give us an API for the membership would like to try it out as well

It's not planned (and I don't even have the bandwidth for that): membership is beyond the scope of OpenIddict, which is an OIDC stack focused on the authorization part. If you need such a thing, consider taking a look at https://github.com/mguinness/IdentityManagerUI.

**Comment by thomasduft** [comment]
Hi there

Maybe my [OpenIddict-UI project](https://github.com/thomasduft/openiddict-ui) that I am about to build could help with this topic. Reviews, ideas and PR's are appreciated as I still fight with a good UX when it comes to manage Scopes and Permissions.

It is built on top of the latest (RC1) OpenIddict Core bits and provides so far API's for managing `Scopes` and `Applications`. The samples then shows the bits in action while providing an Angular client.

I am interested of what you think @kevinchalet ?

Anyway have a nice day, 
Thomas

**Comment by kevinchalet** [comment]
Hey @thomasduft,

> I am interested of what you think @kevinchalet ?

It looks very promising, but if you don't mind, may I ask why you didn't use the OrchardCore OpenID UI?

> Reviews, ideas and PR's are appreciated as I still fight with a good UX when it comes to manage Scopes and Permissions.

If you want to make things a bit less cryptic, you can hide the permissions from the user and simply expose checkboxes allowing to enable the corresponding flows. That's pretty much what we do in OrchardCore.

Cheers.

**Comment by thomasduft** [comment]
Thanks for your answer. 

Regarding the struggles with my UI in my sample client I will have a look at OrchardCore's approach. Thanks for that hint. 

> It looks very promising, but if you don't mind, may I ask why you didn't use the OrchardCore OpenID UI?

First of all, I do not really know OrchardCore. I heard of it when it was a topic on the [.NET Community Standup](https://dotnet.microsoft.com/platform/community/standup) but never really got my hands on it. 

The main reason why I started the project is that it is essentially a port from another repo of mine (see [here](https://github.com/thomasduft/microsts)) that is based on [IdentityServer4](https://identityserver.io/). That repo aims for a really simple but working IdentityProvider that I am using for private "commercial" projects. Based on the announcements of the IdentityServer guys I investigated into the OpenIddict-Core code base and saw some demand for an UI. That's where I thought I could give something back to the community by myself.

I am also not sure whether OrchardCore makes use of the [ASP.NET Core Identity](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/identity) features. I am not saying that this is the best solution, it is just where I put time and effort in in learning it. I use it for managing users.

A more personal reason is that I don't have the satisfying skills of restyling the OrchardCore UI or some other UI in general. In my case this will be an issue. My UI's normally are fairly straight, nothing fancy but functional. The goal of my project won't be to come up with polished ready to use UI that's why I called it "headless". I guess I should this clearly point out somewhere on my readme. All I intend is to demonstrate the functionality the project provides with some sample clients (at the moment an Angular SPA client and eventually also an ASP.NET Razor Page based UI that sits on top of the services the project provides). It is basically to get the idea.

In essence that's it. I hope I could explain you why I investigate in that project.

Cheers, Thomas
